adapter-base/src/main/java/org/eclipse/hono/adapter/auth/device/jwt/JwtAuthProvider.java

```diff
@@ -123,15 +136,14 @@ protected Future<DeviceUser> doValidateCredentials(final JwtCredentials deviceCr
         return result.future();
     }
 
-    private boolean checkJwtValidity(final JwtCredentials deviceCredentials,
+    private Jws<Claims> getJws(final JwtCredentials deviceCredentials,
             final CredentialsObject credentialsOnRecord) {
         try {
             externalJwtAuthTokenValidator.setCredentialsObject(credentialsOnRecord);
-            externalJwtAuthTokenValidator.expand(deviceCredentials.getJwt());
-            return true;
+            return externalJwtAuthTokenValidator.expand(deviceCredentials.getJwt());
         } catch (RuntimeException e) {
-            log.debug("JWT validity check failed", e);
-            return false;
+            log.debug("", e);
+            return null;
```

Comment:
FMPOV we should not swallow the thrown exception but instead catch it in the *doValidateCredentials* method and include it as the *cause* in the *ClientErrorException* that the future is failed with.